File: frontend/grazetrack_app/lib/features/health/models/health_model.dart
class HealthModel {
  final String id;
  final String animalId;
  final String userId;
  final String type;
  final String vaccination;
  final String status;
  final String description;
  final String medicine;
  final double cost;
  final String vet;
  final String nextCheckupDate;
  final String date;
  final String animalType; // e.g. Cow, Goat — used for category filtering

  const HealthModel({
    required this.id,
    required this.animalId,
    required this.userId,
    required this.type,
    this.vaccination = '',
    required this.status,
    this.description = '',
    this.medicine = '',
    this.cost = 0,
    this.vet = '',
    this.nextCheckupDate = '',
    required this.date,
    this.animalType = '',
  });

  factory HealthModel.fromJson(Map<String, dynamic> json) => HealthModel(
        id: json['id'] ?? '',
        animalId: json['animalId'] ?? '',
        userId: json['userId'] ?? '',
        type: json['type'] ?? '',
        vaccination: json['vaccination'] ?? '',
        status: json['status'] ?? 'healthy',
        description: json['description'] ?? '',
        medicine: json['medicine'] ?? '',
        cost: (json['cost'] ?? 0).toDouble(),
        vet: json['vet'] ?? '',
        nextCheckupDate: json['nextCheckupDate'] ?? '',
        date: json['date'] ?? '',
        animalType: json['animalType'] ?? '',
      );
}

File: backend/src/models/healthModel.js
/**
 * Health record data structure for Firestore
 * Collection: health/{healthId}
 */
const HealthModel = {
  animalId: "",          // Reference to animals/{animalId}
  userId: "",            // Who recorded this health event
  type: "",              // vaccination | treatment | checkup | deworming
  vaccination: "",       // Vaccine name (if type is vaccination)
  status: "",            // healthy | sick | recovering | critical
  description: "",       // Description of the health event
  medicine: "",          // Medicine used (if any)
  cost: 0,               // Cost of treatment/medicine
  vet: "",               // Veterinarian name (optional)
  nextCheckupDate: "",   // ISO date for next scheduled checkup
  date: "",              // ISO timestamp of the health event
  createdAt: "",
};

module.exports = HealthModel;
